SDK/include/game_types.hpp

```diff
@@ -0,0 +1,447 @@
+{
+	// todo fill with union
+	uint32_t keys;
+	int16_t upDown;
+	int16_t leftRight;
+};
+
+struct PlayerAnimationData
+{
+	uint16_t ID;
+	uint16_t flags;
+
+	inline Pair<StringView, StringView> name() const
+	{
+		return splitAnimationNames(ID);
+	}
+};
+
+struct PlayerSurfingData
+{
+	enum class Type
+	{
+		None,
+		Vehicle,
+		Object,
+		PlayerObject
+	} type;
+	int ID;
+	Vector3 offset;
+};
+
+/// Holds weapon slot data
+struct WeaponSlotData
+{
+	uint8_t id;
+	uint32_t ammo;
+
+	WeaponSlotData()
+		: id(0)
+		, ammo(0)
+	{
+	}
+
+	WeaponSlotData(uint8_t id)
+		: id(id)
+		, ammo(0)
+	{
+	}
+
+	WeaponSlotData(uint8_t id, uint32_t ammo)
+		: id(id)
+		, ammo(ammo)
+	{
+	}
+
+	uint8_t slot()
+	{
+		static const uint8_t slots[] = {
+			0, // 0 - Fist
+			0, // 1 - Brass Knuckles
+			1, // 2 - Golf Club
+			1, // 3 - Nite Stick
+			1, // 4 - Knife
+			1, // 5 - Baseball Bat
+			1, // 6 - Shovel
+			1, // 7 - Pool Cue
+			1, // 8 - Katana
+			1, // 9 - Chainsaw
+			10, // 10 - Dildo
+			10, // 11 - Dildo
+			10, // 12 - Vibrator
+			10, // 13 - Vibrator
+			10, // 14 - Flowers
+			10, // 15 - Cane
+			8, // 16 - Grenade
+			8, // 17 - Teargas
+			8, // 18 - Molotov Cocktail
+			INVALID_WEAPON_SLOT, //
+			INVALID_WEAPON_SLOT, //
+			INVALID_WEAPON_SLOT, //
+			2, // 22 - Colt 45
+			2, // 23 - Silenced Pistol
+			2, // 24 - Desert Eagle
+			3, // 25 - Shotgun
+			3, // 26 - Sawn-off Shotgun
+			3, // 27 - Combat Shotgun
+			4, // 28 - UZI
+			4, // 29 - MP5
+			5, // 30 - AK47
+			5, // 31 - M4
+			4, // 32 - TEC9
+			6, // 33 - Rifle
+			6, // 34 - Sniper Rifle
+			7, // 35 - Rocket Launcher
+			7, // 36 - Heat Seaker
+			7, // 37 - Flamethrower
+			7, // 38 - Minigun
+			8, // 39 - Satchel Explosives
+			12, // 40 - Bomb
+			9, // 41 - Spray Can
+			9, // 42 - Fire Extinguisher
+			9, // 43 - Camera
+			11, // 44 - Night Vision Goggles
+			11, // 45 - Thermal Goggles
+			11 // 46 - Parachute
+		};
+		if (id >= GLM_COUNTOF(slots))
+		{
+			return INVALID_WEAPON_SLOT;
+		}
+		return slots[id];
+	}
+
+	bool shootable()
+	{
+		return (id >= 22 && id <= 34) || id == 38;
+	}
+};
+
+/// An array of weapon slots
+typedef StaticArray<WeaponSlotData, MAX_WEAPON_SLOTS> WeaponSlots;
+
+enum PlayerWeaponState : int8_t
+{
+	PlayerWeaponState_Unknown = -1,
+	PlayerWeaponState_NoBullets,
+	PlayerWeaponState_LastBullet,
+	PlayerWeaponState_MoreBullets,
+	PlayerWeaponState_Reloading
+};
+
+struct PlayerAimData
+{
+	Vector3 camFrontVector;
+	Vector3 camPos;
+	float aimZ;
+	float camZoom;
+	float aspectRatio;
+	PlayerWeaponState weaponState;
+	uint8_t camMode;
+};
+
+struct PlayerBulletData
+{
+	Vector3 origin;
+	Vector3 hitPos;
+	Vector3 offset;
+	uint8_t weapon;
+	PlayerBulletHitType hitType;
+	uint16_t hitID;
+};
+
+struct PlayerSpectateData
+{
+	enum ESpectateType
+	{
+		None,
+		Vehicle,
+		Player
+	};
+
+	bool spectating;
+	int spectateID;
+	ESpectateType type;
+};
+
+/// The player's name status returned when updating their name
+enum EPlayerNameStatus
```

Comment:
This isn't really a game type

Reply:
True, a better name is probably required.  They're more just generic structs and enums, but I hate the filename `structs.hpp` or similar, it doesn't tell you anything.

Reply:
player_types.hpp or player_structs.hpp?